### PR TITLE
(PDOC-265) Add examples to function reference docs

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by alexjfisher** [comment]
Can you add a test for 4.x API functions too?
Somewhere in here? https://github.com/puppetlabs/puppet-strings/blob/00a0f0e16270b0ce89f8d705847a3ec4dab237d2/spec/unit/puppet-strings/markdown_spec.rb#L118-L125

**Comment by ekohl** [comment]
@alexjfisher like this?

**Comment by alexjfisher** [comment]
@ekohl Not sure...

Where are the examples supposed to go? Above the function, or above the dispatch (or both!)?  Style-guide suggests above the dispatch. https://puppet.com/docs/puppet/5.5/puppet_strings_style.html#example-ruby-function-with-one-potential-return-type

**Comment by alexjfisher** [comment]
ping @hlindberg

**Comment by ekohl** [comment]
I did consider also adding examples to the dispatch functions. That might also make sense for when you use multiple signatures

**Comment by hlindberg** [comment]
It would be nice to be able to write examples above the dispatchers. Sometimes you don't want to do that as the main part shows a progression of examples using different signatures. So, both is preferable IMO.

**Comment by ekohl** [comment]
Updated. It now parses examples within dispatchers/signatures as well. This PR also contains 2 cleanup commits that made it easier for me to understand what was going on but not strictly needed.

File: lib/puppet-strings/yard/handlers/ruby/function_handler.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@ class PuppetStrings::Yard::Handlers::Ruby::FunctionHandler < PuppetStrings::Yard
     required_block_param
     optional_block_param
     return_type
+    example
```
Comment (eputnam) [inline lib/puppet-strings/yard/handlers/ruby/function_handler.rb]: `example` isn't a parameter method, is it? you can remove this.

Comment (ekohl) [reply lib/puppet-strings/yard/handlers/ruby/function_handler.rb]: You're right, it was a debug leftover.

**Comment by ekohl** [comment]
Anything holding back a merge?

**Pull request merged**